autorest/azure/azure.go

```diff
@@ -43,23 +40,64 @@ const (
 )
 
 // ServiceError encapsulates the error response from an Azure service.
+// It adhears to the OData v4 specification for error responses.
 type ServiceError struct {
-	Code    string         `json:"code"`
-	Message string         `json:"message"`
-	Details *[]interface{} `json:"details"`
+	Code    string                   `json:"code"`
+	Message string                   `json:"message"`
+	Details []map[string]interface{} `json:"details"`
```

Comment:
I think there are missing fields
http://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html

Reply:
Yes, this is addressed in a separate PR https://github.com/Azure/go-autorest/pull/227